packages/addon-dev/sample-typescript-rollup.config.mjs

```diff
@@ -0,0 +1,87 @@
+import ts from 'rollup-plugin-ts';
+import { defineConfig } from 'rollup';
+
+import { Addon } from '@embroider/addon-dev/rollup';
+
+const addon = new Addon({
+  srcDir: 'src',
+  destDir: 'dist',
+});
+
+export default defineConfig({
+  // https://github.com/rollup/rollup/issues/1828
+  watch: {
+    chokidar: {
+      usePolling: true,
+    },
+  },
+  output: {
+    ...addon.output(),
+    sourcemap: true,
+    // Until a bug with the ember-cli-htmlbars-removal transform is fixed, you
+    // may need
+    // hoistTransitiveImports: false
+    // specified here as well
+  },
+  plugins: [
+    // These are the modules that users should be able to import from your
+    // addon. Anything not listed here may get optimized away.
+    addon.publicEntrypoints([
+      'index.ts',
+      'components/**/*.ts',
+      'helpers/**/*.ts',
+      'modifiers/**/*.ts',
+      'services/**/*.ts',
+      'test-support/**/*.ts',
+    ]),
+
+    // These are the modules that should get reexported into the traditional
+    // "app" tree. Things in here should also be in publicEntrypoints above, but
+    // not everything in publicEntrypoints necessarily needs to go here.
+    addon.appReexports([
```

Comment:
as an aside, is there anything we can do with `appReexports` and `publicEntrypoints` to either not need file extensions, or have some sort of assertions on over-specification of extensions? The plugins are very forgiving here, and I'm not sure what's "proper".